src/lib/statistics/generate_pruning_statistics.cpp

```diff
@@ -86,15 +92,21 @@ void generate_chunk_pruning_statistics(const std::shared_ptr<Chunk>& chunk) {
 
 void generate_chunk_pruning_statistics(const std::shared_ptr<Table>& table) {
   const auto chunk_count = table->chunk_count();
+  auto jobs = std::vector<std::shared_ptr<AbstractTask>>{};
+  jobs.reserve(chunk_count);
+
   for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
     const auto chunk = table->get_chunk(chunk_id);
 
     if (!chunk || chunk->is_mutable()) {
```

Comment:
I think it's worth to add the condition from L50 also here. Otherwise, we schedule a task that does a no-op for each chunk when adding the table to the storage manager after encoding.

Reply:
How to deal with the fact that L50 has a comment? I'm always a bit reluctant of copying comments, because they get out of sync so easily

Reply:
I agree. Do we ever call the version for a single chunk directly? And if so, one could argue that the caller could (should) do the check. So I'm fine with moving both the check and the comment to here.

Reply:
The single-chunk call is the one done in the ChunkEncoder (when we "correctly parallelize").

Reply:
O sure, we call this e.g. in the chunk encoder. I guess then it's fine to add the check there as well.

Reply:
So I extracted the check into a separate method. Thoughts?